test/extended/util/operator.go

```diff
@@ -91,7 +94,31 @@ func waitForOperatorProgressingToBe(ctx context.Context, configClient configv1cl
 	_, err := watchtools.UntilWithSync(ctx,
 		cache.NewListWatchFromClient(configClient.ConfigV1().RESTClient(), "clusteroperators", "", fields.Everything()),
 		&configv1.ClusterOperator{},
-		nil,
+		func(store cache.Store) (bool, error) {
+			obj, exists, err := store.Get(&metav1.ObjectMeta{Namespace: "", Name: operatorName})
+			if err != nil {
+				return true, err
+			}
+			if !exists {
+				return false, kerrors.NewNotFound(schema.GroupResource{
+					Group:    "config.openshift.io/v1",
+					Resource: "ClusterOperator",
+				}, operatorName)
+			}
+			operator := obj.(*configv1.ClusterOperator)
+			if desiredProgressing {
+				if v1helpers.IsStatusConditionTrue(operator.Status.Conditions, configv1.OperatorProgressing) {
+					return true, nil
+				}
+				return false, nil
+			}
+
+			if v1helpers.IsStatusConditionFalse(operator.Status.Conditions, configv1.OperatorProgressing) {
+				return true, nil
```

Comment:
In the case that the state we want to reach is `progressing=false`, we do a check for `available=true` and `degraded=false` as well.

Reply:
> I don't think the degraded check was doing what you thought it was doing. You need to be sure the level is achieved, but I wouldn't expect a degraded on a two minute momentum to have a meaningful impact for you

We can move forward without a check for degraded, but our working assumption is that tests on a degraded state are done and fail.
The inertia is set to 30 minutes in our case and not 2 minutes.

Reply:
Ok, we might not need a degraded check at all. But we would need to check available=true alongside progressing=false, right?